scrubadub/detectors/named_entity.py

```diff
@@ -0,0 +1,60 @@
+from typing import Dict, Generator, Iterable, Optional, Sequence, Union
+
+import spacy
+
+from .base import Detector
+from ..filth import NamedEntityFilth, Filth, NameFilth, OrganizationFilth
+from ..utils import CanonicalStringSet
+
+
+class NamedEntityDetector(Detector):
+    """Use spacy's named entity recognition to clean named entities.
+     List specific entities to include passing ``named_entities``, e.g.
+     (PERSON)
+    """
+    filth_cls_map = {
+        'PERSON': NameFilth,
+        'ORG': OrganizationFilth
+    }
+    name = 'named_entity'
+
+    disallowed_nouns = CanonicalStringSet(["skype"])
+
+    def __init__(self, named_entities: Iterable[str] = {'PERSON'},
+                 model: str = "en_core_web_trf", **kwargs):
+        # Spacy NER are all upper cased
+        self.named_entities = {entity.upper() for entity in named_entities}
+        if model not in spacy.info()['pipelines']:
+            raise OSError("Can't find model '{}'. If it is a valid Spacy model, "
+                          "download it (e.g. with the CLI command "
+                          "`python -m spacy download {}`).".format(model, model))
+        self.nlp = spacy.load(model)
+        # Only enable necessary pipes
+        self.nlp.select_pipes(enable=["transformer", "tagger", "parser", "ner"])
+        super(NamedEntityDetector, self).__init__(**kwargs)
+
+    def _iter_spacy_pipeline(self, doc_names: Sequence[Optional[str]], doc_list: Sequence[str]):
+        for doc_name, doc in zip(doc_names, self.nlp.pipe(doc_list)):
+            for ent in doc.ents:
+                if ent.label_ in self.named_entities:
+                    # If there is no standard 'filth', returns a NamedEntity filth
+                    filth_cls = self.filth_cls_map.get(ent.label_, NamedEntityFilth)
+                    yield filth_cls(beg=ent.start_char,
+                                    end=ent.end_char,
+                                    text=ent.text,
+                                    document_name=(str(doc_name) if doc_name else None),  # None if no doc_name provided
+                                    detector_name=self.name,
+                                    label=ent.label_)
+
+    def iter_filth_documents(self, documents: Union[Sequence[str], Dict[str, str]]) -> Generator[Filth, None, None]:
```

Comment:
Since this function is only boilerplate and would have to be repeated in each of the detectors, we could make the interface more similar to the `_iter_spacy_pipeline` function and keep this part in the scrubber, perhaps?

This would mean that the signature for iter_filth_documents would look like this:

```def iter_filth_documents(self, doc_names: Sequence[Optional[str]], doc_list: Sequence[str]) -> Generator[Filth, None, None]:```

Does that sound like a good idea?